src/vm_x86.dasc

```diff
@@ -124,16 +124,46 @@
 |.define CFRAME_SPACE,	aword*7			// Delta for esp (see <--).
 |.macro saveregs_
 |  push edi; push esi; push ebx
+|.if WIN
+|  push 0
+|  fs; push dword [0]
+|.endif
 |  sub esp, CFRAME_SPACE
 |.endmacro
 |.macro saveregs
 |  push ebp; saveregs_
 |.endmacro
 |.macro restoreregs
 |  add esp, CFRAME_SPACE
+|.if WIN
+|  fs; pop dword [0]
+|  pop edi					// Just to increment esp by 4
+|.endif
 |  pop ebx; pop esi; pop edi; pop ebp
 |.endmacro
 |
+|.if WIN
```

Comment:
My understanding was that 16-byte alignment should be preserved on x64, and on non-Windows x86, but that Windows x86 only maintained 4-byte stack alignment. Agner's calling conventions guide confirms this understanding, but conservative 16-byte alignment is nevertheless an option.